### PR TITLE
add leetcode Lowest Common Ancestor of a Binary Tree

### PR DESCRIPTION
**Pull request by alexpantyukhin**
#### Description of Change

Add a guide for new LeetCode solutions.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

File: leetcode/src/236.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ *     struct TreeNode *right;
+ * };
+ */
+
+struct List {
+    struct TreeNode* node;
+    struct List* next;
+};
+
+bool findPathOfTarget(struct TreeNode* node, struct TreeNode* target, struct List* path){
```
Comment (tjgurwara99) [inline leetcode/src/236.c:15]:
```suggestion
bool findTargetPath(struct TreeNode* node, struct TreeNode* target, struct List* path){
```

Avoid prepositions when you can 😄 they almost always add no real value when reading code.

File: leetcode/src/236.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * struct TreeNode {
+ *     int val;
+ *     struct TreeNode *left;
+ *     struct TreeNode *right;
+ * };
+ */
+
+struct List {
+    struct TreeNode* node;
+    struct List* next;
```
Comment (tjgurwara99) [inline leetcode/src/236.c:12]: Could you add a comment explaining why you added this? I believe you don't need an array of `List` for this usually (unless I'm mistaken) so documenting why you added it is better for learning 😄

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]
Just some minor comment, otherwise very clearly written program.

File: leetcode/src/236.c (hunk 2 above)
Comment (alexpantyukhin) [reply leetcode/src/236.c:12]: Yes, sure)) probably it should be `ListItem`. Sorry)

**Comment by alexpantyukhin** [comment]
Review notes have been fixed.

**Review by tjgurwara99 — DISMISSED** [review]

File: leetcode/src/236.c [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ *     int val;
+ *     struct TreeNode *left;
+ *     struct TreeNode *right;
+ * };
+ */
+
+// The list for TreeNodes.
+struct ListItem {
+    struct TreeNode* node;
+    struct ListItem* next;
```
Comment (Panquesito7) [inline leetcode/src/236.c:13]: Please add a one-line description of what the variable is for (**see the example below**).

```c
int64_t data = 0;       ///< The value/key of the node.
```

Comment (alexpantyukhin) [reply leetcode/src/236.c]: fixed

File: leetcode/src/236.c [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    if (target->next != NULL){
+        freeList(target->next);
+    }
+    
+    free(target);
+}
+
+
+// Find full path for p and q.
+// Find the longest common path in pathes.
```
Comment (tjgurwara99) [inline leetcode/src/236.c:53]:
```suggestion
// Find the longest common path in paths.

```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

**Review by tjgurwara99 — APPROVED** [review]

**Review by Panquesito7 — APPROVED** [review]

**Pull request merged**